File: LibraryApp/LibraryApp.Database/Entities/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryApp.Database.Entities
{
    public class Book
    {
        public int Id { get; set; }

        [Required]
        public required string Title { get; set; }

        public string? Description { get; set; }

        [Required]
        public required string Author { get; set; }
    }
}

File: library-ui/src/classes/book.ts
export default class Book {
  id = 0
  title = ''
  description = ''
  author = ''

  constructor (build: any | undefined | null = undefined) {
    if (build != null && build != undefined) {
      this.id = build.id
      this.title = build.title
      this.description = build.description
      this.author = build.author
    }
  }
}
